### PR TITLE
build_config: Include vars in build config version

### PR DESCRIPTION
**Pull request by justincampbell**

### CODE REVIEW

File: v1/build_config.go [hunk 1]
```diff
@@ -179,5 +180,6 @@ type bcCreateWrapper struct {
 	Version struct {
 		Metadata map[string]interface{} `json:"metadata,omitempty"`
 		Builds   []BuildConfigBuild     `json:"builds"`
+		Vars     map[string]string      `json:"vars"`
```
Comment (pearkes) [inline v1/build_config.go]: Should this be omitempty?

**Comment by pearkes** [comment]
:+1:

File: v1/build_config.go (hunk 1 above)
Comment (justincampbell) [inline v1/build_config.go]: Done! I think this is good because it forces Atlas to handle old and new clients, instead of always expecting a value there.

**Comment by justincampbell** [comment]
Failure seems separate from these changes. Ok to merge? @pearkes @ryanuber @sethvargo

**Comment by cbednarski** [comment]
@justincampbell Check go vet:

> v1/build_config.go:183: struct field tag `json:"vars",omitempty` not compatible with reflect.StructTag.Get

File: v1/build_config.go [hunk 2]
```diff
@@ -179,5 +180,6 @@ type bcCreateWrapper struct {
 	Version struct {
 		Metadata map[string]interface{} `json:"metadata,omitempty"`
 		Builds   []BuildConfigBuild     `json:"builds"`
+		Vars     map[string]string      `json:"vars",omitempty`
```
Comment (sethvargo) [inline v1/build_config.go]: `omitempty` goes inside quotes:

```go
`json:"vars,omitempty"`
```

**Comment by justincampbell** [comment]
@mwhooker Thanks!

**Comment by mwhooker** [comment]
any reservations against merging this?

**Comment by cbednarski** [comment]
Looks good to me. Callers will need to update `UploadBuildConfigVersion`.

File: v1/build_config.go [hunk 3]
```diff
@@ -126,7 +126,7 @@ func (c *Client) CreateBuildConfig(user, name string) (*BuildConfig, error) {
 //
 // Actual API: "Create Build Config Version"
 func (c *Client) UploadBuildConfigVersion(v *BuildConfigVersion, metadata map[string]interface{},
-	data io.Reader, size int64) error {
+	vars map[string]string, data io.Reader, size int64) error {
```
Comment (sethvargo) [inline v1/build_config.go]: Could we borrow a book from the aws-sdk-go here and make this a struct input instead? That will give us the flexibility to add new fields in the future without breaking the signature.

Comment (justincampbell) [inline v1/build_config.go]: Agreed that would be better. Would like to do it outside of the scope of this PR though, and for consistency we would make the same breaking change to all other functions.

**Pull request merged**